tests/test_hypothesis.py

```diff
@@ -0,0 +1,186 @@
+        mocked_filter_dates.return_value = ["filtered_path"]
+        mocked_read_csv.return_value = mock_df
+
+        hypothesis_instance = Hypothesis()
+        df = hypothesis_instance.get_covid_data(time_period, location)
+        formatted_date = df.iloc[0]["Date_of_statistics"].strftime("%Y-%m-%d")
+        assert not df.empty
+        assert df.iloc[0]["Version"] == 2
+        assert df.iloc[0]["Date_of_report"] == "2022-01-03T09:00:00Z"
+        assert formatted_date == "2022-01-01"
+        assert df.iloc[0]["Security_region_code"] == "VR01"
+        assert df.iloc[0]["Security_region_name"] == "Groningen"
+        assert df.iloc[0]["Tested_with_result"] == 976
+        assert df.iloc[0]["Tested_positive"] == 378
+
+        # assert mock methods
+        mocked_get_file_paths.assert_called_once()
+        mocked_filter_dates.assert_called_once_with(
+            mocked_get_file_paths.return_value, time_period
+        )
+        mocked_read_csv.assert_called_once_with(
+            "https://github.com/mzelst/covid-19/raw/"
+            "master/data-rivm/tests/filtered_path",
+            compression="gzip",
+            header=0,
+            sep=",",
+            quotechar='"',
+        )
+
+    def test_location_not_in_dataset(self):
+        # trying to get data for an unknown location
+        # should raise an error
+        with pytest.raises(ValueError):
+            Hypothesis.get_positive_cases(
+                10, ("2023-01-01", "2023-01-03"), "Los Angeles"
+            )
+
+    def test_scaling(self):
+        # Tests the scale_cases_to_population function
+        real_size = 100
+        sim_size = 50
+        daily_positive_cases = pd.Series([10, 20, 30, 40, 50])
+        out = Hypothesis.scale_cases_to_population(
+            daily_positive_cases, real_size, sim_size
+        )
+        expected = pd.Series([5, 10, 15, 20, 25])
+        assert all(out == expected), f"Expected {expected}, but got {out}"
+
+    def test_scaling_warning(self):
+        # Tests the warning of scale_cases_to_population function
+        daily_positive_cases = pd.Series([1, 1, 1, 1, 1])
+        real_size = 1000
+        sim_size = 50
+        with warnings.catch_warnings(record=True) as w:
+            Hypothesis.scale_cases_to_population(
+                daily_positive_cases, real_size, sim_size
+            )
+            # is there warning there?
+            assert len(w) == 1, f"Expected warning, but got {len(w)} warnings"
+
+    def test_no_scaling_warning(self):
+        # Tests the warning of scale_cases_to_population function
+        # I expect no warning here
+        daily_positive_cases = pd.Series([500, 600, 700, 800, 900])
+        real_size = 1000
+        sim_size = 500
+        with warnings.catch_warnings(record=True) as w:
+            Hypothesis.scale_cases_to_population(
+                daily_positive_cases, real_size, sim_size
+            )
+            assert len(w) == 0, f"Expected no warnings, but got {len(w)}"
```

Comment:
As above, I would try to have a single, parametrized test. Here, the expected warnings could be a parameter too.